lua/starfall/libs_sh/vectors.lua

```diff
@@ -115,6 +116,15 @@ function vec_meta.__mul(a, b)
 		return wrap({ b[1] * a, b[2] * a, b[3] * a })
 	elseif dgetmeta(a) == vec_meta and dgetmeta(b) == vec_meta then
 		return wrap({ a[1] * b[1], a[2] * b[2], a[3] * b[3] })
+	elseif dgetmeta(a) == vec_meta and dgetmeta(b) == quat_meta then -- Vector * Quaternion
```

Comment:
I think should be Quaternion * Vector

Reply:
`Quaternion * Vector` already exists as a part of the Quaternion multiplication.
`Vector * Quaternion` performs the same exact calculations so I guess it's valid and could be left here, not sure if there's an actual use for it when working with quaternions though.
Or am I missing a point on what you're saying?

Reply:
I just had a look at the math in this and I don't even know what its doing. I'd assume you want to rotate the vector but this doesn't do that.

Reply:
Right, I think I get it now, it creates a new quaternion from the vector, which should not be the case.
`Quaternion * Vector` is working good I think, since it attempts to rotate the Quaternion, not the Vector.
Not sure if I should completely removed it or leave it as an alias for `Vector:getQuaternion()` (`Quaternio(0, vec.x, vec.y, vec.z`) which could be handy, what do you think?

Reply:
Is that what it does? rotates Q around axis V? Probably need to check that math then.

Reply:
Assuming that E2 implementation is correct:
https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/quaternion.lua#L353

If it's not, the I guess this should be used:
https://gamedev.stackexchange.com/a/50545
Another source for the same calculations (this time in Lua's __mul):
https://github.com/bjornbytes/maf/blob/master/maf.lua#L177

Again, I have no idea how quaternions work, so I'm pretty much just looking at multiple random implementations of them.

Reply:
I don't think the __mul should behave differently depending on which side the vector is. It should either rotate the vector or rotate the quaternion. Rotating the vector makes more sense to me.

Reply:
Welp, I don't know the maths behind rotating vector by a quaternion, so you're welcome to do it :)
Please don't try explaining quaternions to me...

Reply:
It behaves differently because quaternion-quaternion multiplication is non-commutative and vectors are interpreted as quaternions with the real part 0. Mathematically, to rotate a vector by a quaternion you calculate `q*v*p:conjugate()` and extract the vector part.

Reply:
@CornerPin I've implemented (or at least attempted) your solution. It seems like it's working fine, but would appreciate if you checked it.
The following code rotates the hologram around a chip `twice` per `one full rotation` on either `pitch` or `roll`. Yaw doesn't cause the hologram to go to infinity, it does nothing from what I can see.
```lua
local holo = holograms.create(Vector(), Angle(), "models/holograms/cube.mdl")

hook.add("tick", "", function()
    local up = chip():getUp()
    local q = chip():getQuaternion()
    local rot = up * q * 20
    
    holo:setPos(chip():getPos() + rot)
end)
```

Reply:
The code works as it should, but if you try to raise the quaternion to a number power, it results in a non-unit quaternion.
```lua
local holo = holograms.create(Vector(), Angle(), "models/holograms/cube.mdl")

hook.add("tick", "", function()
    local up = chip():getUp()
    local q = chip():getQuaternion()
    local rot = up * q^(3 - 1) * 20 -- Should rotate three times the angle, but it doesn't
    
    holo:setPos(chip():getPos() + rot)
end)
```
The reason why it happens is that `__pow(Quaternion, number)` doesn't call the exponential function before returning the quaternion.

Reply:
Thanks! Must've missed that when doing *\*cough*\* *yoinking* *\*cough*\* mathz.